src/types/closure.md

```diff
@@ -412,37 +416,55 @@ Output is the same: `C' = C`
 When you have a closure that both references a packed field (which is unsafe) and moves from it (which is safe) we capture the entire struct, rather than just moving the field. This is to aid in predictability, so that removing the move doesn't make the closure become unsafe:
 
 ```rust
-print(&packed.x);
-move_value(packed.x);
+#[repr(packed)]
+struct Packed { x: String }
+
+# fn use_ref<T>(_: &T) {}
+# fn move_value<T>(_: T) {}
+
+fn main() {
+    let packed = Packed { x: String::new() };
+
+    let c = || {
+        use_ref(&packed.x);
+        move_value(packed.x);
+    };
+
+    c();
+}
 ```
 
+```
+Closure mode = ref
+C = {
+    (ref mut, packed)
+}
+C' = C
+```
 
+### Optimization-Edge-Case
 ```rust
-struct Point { x: i32, y: i32 }
-fn f(p: &Point) -> impl Fn() {
-    let c = move || {
-      let x = p.x; 
-    }; 
-    
-    // x.x -> ByValue
-    // after rules x -> ByValue
+struct Int(i32);
+struct B<'a>(&'a i32);
 
+struct MyStruct<'a> {
+   a: &'static Int,
+   b: B<'a>,
+}
+
+fn foo<'a, 'b>(m: &'a MyStruct<'b>) -> impl FnMut() + 'static {
+    let c = || drop(&m.a.0);
     c
-} 
+}
 
-struct Point { x: i32, y: i32 }
```

Comment:
These examples were us discussiong proper result of what analysis should do in moving copy types and working through some exampels